### PR TITLE
added test case for empty name argument

### PR DESCRIPTION
**Pull request by sallyom**
@nhr and @derekwaynecarr please review.

### CODE REVIEW

File: plugins/route/allocation/simple/plugin_test.go [hunk 1]
```diff
@@ -39,13 +39,24 @@ func TestNewSimpleAllocationPlugin(t *testing.T) {
...
-		_, err := NewSimpleAllocationPlugin(tc.Name)
+		sap, err := NewSimpleAllocationPlugin(tc.Name)
 		if err != nil && !tc.ErrorExpectation {
 			t.Errorf("Test case for %s got an error where none was expected", tc.Name)
 		}
+		if len(tc.Name) > 0 {
+			continue
+		}
+		dap := &SimpleAllocationPlugin{DNSSuffix: defaultDNSSuffix}
+		if len(tc.Name) == 0 && (sap.DNSSuffix != dap.DNSSuffix) {
```
Comment (derekwaynecarr) [inline plugins/route/allocation/simple/plugin_test.go]: The len check here is unnecessary given your len check above.

Remove one of them ;-)

**Comment by sallyom** [comment]
[test]

**Comment by derekwaynecarr** [comment]
[merge]

**Comment by derekwaynecarr** [comment]
squash commits and i will merge.

ping me if you need help on that.

**Comment by derekwaynecarr** [comment]
[merge]

**Pull request merged**